pandas/tests/frame/methods/test_quantile.py

```diff
@@ -14,6 +14,19 @@
 
 
 class TestDataFrameQuantile:
+    def test_numeric_only_default_false_warning(self):
+        # GH #7308
+        df = DataFrame({"A": [1, 2, 3], "B": [2, 3, 4]})
+        df["C"] = pd.date_range("2014-01-01", periods=3, freq="m")
+
+        df_expected_num_only_true = Series(
+            [2.0, 3.0],
+            index=["A", "B"],
+            name=0.5,
+        )
+        with tm.assert_produces_warning(FutureWarning, match="numeric_only"):
+            tm.assert_series_equal(df.quantile(0.5), df_expected_num_only_true)
```

Comment:
Please do ``result = df.quantile(0.5)`` on a separate line so that the series equal is not within the with statement